x-pack/plugins/upgrade_assistant/README.md

```diff
@@ -28,22 +28,27 @@ These surface runtime deprecations, e.g. a Painless script that uses a deprecate
 
 ### Fixing problems
 
 #### Elasticsearch
 
 Elasticsearch deprecations can be handled in a number of ways:
 
-* **Reindexing.** When a user's index contains deprecations (e.g. mappings) a reindex solves them. Currently, the Upgrade Assistant only automates reindexing for old indices. For example, if you are currently on 7.x, and want to migrate to 8.0, but you still have indices that were created on 6.x. For this scenario, the user will see a "Reindex" button that they can click, which will perform the reindex.
-  * Reindexing is an atomic process in Upgrade Assistant, so that ingestion is never disrupted.
-    It works like this:
-    * Create a new index with a "reindexed-" prefix ([#30114](https://github.com/elastic/kibana/pull/30114)).
-    * Create an index alias pointing from the original index name to the prefixed index name.
-    * Reindex from the original index into the prefixed index.
-    * Delete the old index and rename the prefixed index.
-Currently reindexing deprecations are only enabled for major version upgrades  by setting the config `featureSet.reindexCorrectiveActions` to `true` on the `x.last` version of the stack.
+* **Reindexing.** When a user's index contains deprecations (e.g. mappings) a reindex solves them. The Upgrade Assistant only automates reindexing for indices. For example, if you are currently on 7.x, and want to migrate to 8.0, but you still have indices that were created on 6.x. For this scenario, the user will see a "Reindex" button that they can click, which will perform the reindex.
+  * Reindexing is an idempotent action in Upgrade Assistant. It works like this ([overview in code](https://github.com/elastic/kibana/blob/b320a37d8b703b2fa101a93b6971b36ee2c37f06/x-pack/plugins/upgrade_assistant/server/lib/reindexing/reindex_service.ts#L498-L540)):
+    1. Set a write-block on the original index, no new data can be written during reindexing.
+    2. Create a target index with the following name `reindexed-v{majorVersion}-{originalIndex}`. E.g., if `my-index` is the original, the target will be named `reindexed-v8-my-index`.
+    3. [Reindex](https://www.elastic.co/guide/en/elasticsearch/reference/master/docs-reindex.html) from the original index to the target index. Kibana will continuously report reindexing status.
+    4. Once reindexing is done, create an alias from the original index to the target index. All existing aliases referencing the original index will be re-pointed to the target index. E.g., `my-index` will be an alias referencing `reindexed-v8-my-index`.
+       1. **NOTE:** writing/indexing will effectively be re-enabled at this point via the alias, unless the original was write-blocked by users as index settings are inherited from the original.
+    5. Delete the original index.
```

Comment:
Not sure how much the average reader of this doc cares about this level of detail but I thought it might remove any anxiety / doubts about potentially having an alias point to to two indices or other kinds of race conditions

```suggestion
    4. Once reindexing is done, in one atomic operation via the aliases API:
      1. Create an alias from the original index to the target index. All existing aliases referencing the original index will be re-pointed to the target index. E.g., `my-index` will be an alias referencing `reindexed-v8-my-index`.
      2. Delete the original index.
      3. **NOTE:** writing/indexing will effectively be re-enabled at this point via the alias, unless the original was write-blocked by users as index settings are inherited from the original.
```

Reply:
I like it!